bin/echo.c

```diff
@@ -28,6 +28,48 @@
 
 #include <s2n.h>
 
+#include "crypto/s2n_rsa.h"
+#include "stuffer/s2n_stuffer.h"
+#include "utils/s2n_safety.h"
+
+int accept_all_rsa_certs(struct s2n_blob *client_cert_chain_blob, struct s2n_cert_public_key *cert_public_key, void *context)
+{
+    struct s2n_stuffer cert_chain_in;
+    GUARD(s2n_stuffer_init(&cert_chain_in, client_cert_chain_blob));
+    GUARD(s2n_stuffer_write(&cert_chain_in, client_cert_chain_blob));
+
+    int certificate_count = 0;
+        while (s2n_stuffer_data_available(&cert_chain_in)) {
+            uint32_t certificate_size;
+
+            GUARD(s2n_stuffer_read_uint24(&cert_chain_in, &certificate_size));
+
+            if (certificate_size > s2n_stuffer_data_available(&cert_chain_in) || certificate_size == 0) {
```

Comment:
s2n_stuffer_read_uint24 will read 3 bytes from the stuffer than contains the length of the entire Cert Chain. Here we are validating that the amount of data in the stuffer is actually enough to read the entire chain.